### PR TITLE
Remove DecompilerPrinter.scala

### LINKED ISSUE #14806: Remove old DecompilerPrinter
The old implementation for the decompiler `dotty.tools.dotc.printing.DecompilerPrinter` is not used by the decompiler anymore. It is only used by the `ReplPrinter`. We should remove `DecompilerPrinter` by merging it with `ReplPrinter`.

### PR DESCRIPTION
**Pull request by EstebanMarin**
Description: We tested all methods in REPL and none of them the methods being used.
All tests pass. `do while` loop are not supported anymore. `packages` are not printed in REPL
nor content of classes either.

fixes #14806

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/printing/ReplPrinter.scala [hunk 1]
```diff
@@ -11,8 +11,10 @@ import dotty.tools.dotc.core.Types._
 import dotty.tools.dotc.printing.Texts._
 
 
-class ReplPrinter(_ctx: Context) extends DecompilerPrinter(_ctx) {
+class ReplPrinter(_ctx: Context) extends RefinedPrinter(_ctx) {
 
+  //try to see what is use for for the code and the REPL
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/printing/ReplPrinter.scala:16]: We should remove this comment

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

**Comment by nicolasstucki** [comment]
To fix the MiMa tests, we should be able to re-run the tests after  #15457 is merged (no-rebase needed).

**Comment by nicolasstucki** [comment]
@EstebanMarin could you squash the commits?

**Comment by EstebanMarin** [comment]
@nicolasstucki done

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**